### ISSUE TITLE
Edit card error

### ISSUE THREAD

**Issue by buituan11** [rcdexta/react-trello#424]
**Describe the bug**
I see source code and i found some bugs like this
when i update a card it will pass params like this 
` updateCard = updatedCard => {
    this.props.actions.updateCard({laneId: this.props.id, updatedCard})
    this.props.onCardUpdate(this.props.id, updatedCard)
  }`
then it calls to LaneHelper like this 
` updateCardFromLane: (state, {laneId, card}) => {
...
}`
Therefore card is undefined. It must be 
` updateCardFromLane: (state, {laneId, updatedCard}) => {
...
}`
Can you fix this bug ?? Thanks

**Comment by bradgreens** [comment]
I ran into this as well wit h`"react-trello": "^2.2.9"`. 

1. Start with a blank board
2. Add a lane, add a card
3. Modify the title of the card and blur the card to update it
4. Observe `TypeError: Cannot read property 'id' of undefined`
4a. Observe in screenshot, the `updateCardFromLane` function is called from the reducer, but the reducer has passed along `updatedCard` instead of `card` in the destructured function args.

I was able to hack/fix it on my machine by changing https://github.com/rcdexta/react-trello/blob/master/src/controllers/Lane.js#L145:

`this.props.actions.updateCard({laneId: this.props.id, updatedCard})`

to

`this.props.actions.updateCard({laneId: this.props.id, updatedCard, card: updatedCard}) // map updatedCard to expected card argument`.

I assume there is probably a more complete way to address the issue, but thought I'd share.

**Comment by bradgreens** [comment]
I noticed that partial updates to the card were overwriting the properties which were not being edited. e.g. Updating a title would wipe the description. I tweaked my hack to the following:
```
    const lane = state.lanes[laneIndex];
    const cardIndex = lane.cards.findIndex(x => x.id === card.id);

    if (cardIndex < 0) {
      return state;
    }
  
    // map partial updates onto the previous card state
    const prevCard = lane.cards[cardIndex];
    const newCard = { ...prevCard, ...card };
    
    return // ... omitted 
```

**Comment by WolfyUK** [comment]
Thanks, @bradgreens. I've included both of your fixes in the following `react-trello+2.2.9.patch` file, if anybody else happens to be using [patch-package](https://www.npmjs.com/package/patch-package).

```diff
@@ -0,0 +1,44 @@
diff --git a/node_modules/react-trello/dist/controllers/Lane.js b/node_modules/react-trello/dist/controllers/Lane.js
index 56c033c..a425d10 100644
--- a/node_modules/react-trello/dist/controllers/Lane.js
+++ b/node_modules/react-trello/dist/controllers/Lane.js
@@ -172,7 +172,8 @@ class Lane extends _react.Component {
     (0, _defineProperty2.default)(this, "updateCard", updatedCard => {
       this.props.actions.updateCard({
         laneId: this.props.id,
-        updatedCard
+        // https://github.com/rcdexta/react-trello/issues/424#issuecomment-764289969
+        card: updatedCard
       });
       this.props.onCardUpdate(this.props.id, updatedCard);
     });
diff --git a/node_modules/react-trello/dist/helpers/LaneHelper.js b/node_modules/react-trello/dist/helpers/LaneHelper.js
index 358d201..790cd32 100644
--- a/node_modules/react-trello/dist/helpers/LaneHelper.js
+++ b/node_modules/react-trello/dist/helpers/LaneHelper.js
@@ -149,18 +149,24 @@ const LaneHelper = {
       return state;
     }
 
+    // https://github.com/rcdexta/react-trello/issues/424#issuecomment-765104545
+    const lane = state.lanes[laneIndex];
     const cardIndex = state.lanes[laneIndex].cards.findIndex(x => x.id === card.id);
 
     if (cardIndex < 0) {
       return state;
     }
 
+    // map partial updates onto the previous card state
+    const prevCard = lane.cards[cardIndex];
+    const newCard = { ...prevCard, ...card };
+
     return (0, _immutabilityHelper.default)(state, {
       lanes: {
         [laneIndex]: {
           cards: {
             [cardIndex]: {
-              $set: card
+              $set: newCard
             }
           }
         }
```
